source/core/utilities.cc

```diff
@@ -380,3 +380,106 @@ make_table_tensors_scalars(
+          return_value = std::string(matches[1].first, matches[1].second);
+        }
+    }
+
+  return return_value;
+}
+
+unsigned int
+get_dimension(const std::string &file_name)
+{
+  const std::string dimension =
+    get_last_value_of_parameter(file_name, "dimension");
+
+  if (dimension.size() > 0)
+    {
+      // Extracted from ASPECT
+      // A common problem is that people have .prm files that were generated
+      // on Windows, but then run this on Linux where the line endings are
+      // different. This is pernicious because it means that the conversion
+      // of a string such as "2\r" to an integer fails, but if we print
+      // this string, it comes out completely garbled because it contains
+      // a carriage-return without a newline -- so the error message looks
+      // like this:
+      //
+      //    >.  While reading the dimension from the input file, ASPECT found a
+      //    string that can not be converted to an integer: <2
+      //
+      // Note how the end of the error message overwrites the beginning
+      // of the line.
+      //
+      // To avoid this kind of error, specifically test up front that the
+      // text in question does not contain '\r' characters. If we are on
+      // linux, then this kind of character would means that the line endings
+      // are wrong. On the other hand, if we are on windows, then the
+      // getline command we have used in finding 'dimension' would have
+      // filtered it out. So its presence points to a problem.
+
+      AssertThrow(
+        dimension.find('\r') == std::string::npos,
+        dealii::ExcMessage(
+          "It appears that your input file uses Windows-style "
+          "line endings ('\\r\\n') but you are running on a system where "
+          "the C++ run time environment expects input files to have "
+          "Unix-style line endings ('\\n'). You need to convert your "
+          "input file to use the correct line endings before running "
+          "ASPECT with it."));
+      try
+        {
+          return dealii::Utilities::string_to_int(dimension);
+        }
+      catch (...)
+        {
+          AssertThrow(false,
+                      dealii::ExcMessage(
+                        "While reading the dimension from the input file, "
+                        "Lethe found a string that can not be converted to "
+                        "an integer: <" +
+                        dimension + ">."));
+          return 0; // we should never get here.
+        }
+    }
+  else
+    {
+      AssertThrow(
+        false,
+        dealii::ExcMessage(
+          "While reading the dimension from the input file, "
+          "Lethe found a value that is neither 2 or 3. Since August 2023, "
+          "Lethe requires that the user explicitely specify the dimension of the problem within the parameter file. This can be achieved by adding set dimension = 2 or set dimension = 3 within the parameter file"));
```

Comment:
```suggestion
          "Lethe requires that the user explicitly specify the dimension of the problem within the parameter file. This can be achieved by adding set dimension = 2 or set dimension = 3 within the parameter file"));
```